### ISSUE TITLE
[BUG] model_selection/split does not support y being an index since 0.8.1

### ISSUE THREAD

**Issue by hoesler4dh** [alan-turing-institute/sktime#1580]
**Describe the bug**
#1456 introduced a bug. If y is an Index, the code leads to an unassigned local variable
```
UnboundLocalError: local variable 'y_index' referenced before assignment
```

**To Reproduce**
```python
from sktime.forecasting.model_selection import SingleWindowSplitter
import pandas as pd
next(SingleWindowSplitter(fh=1).split(pd.PeriodIndex(year=[2000, 2002], quarter=[1, 3])))
```

**Expected behavior**
split function should accept an Index as argument

**Versions**
0.8.1

**Comment by fkiraly** [comment]
ouch. Thanks for reporting, @hoesler4dh!

Well, I would put this down as a consequence to no one writing proper docstrings.

I was unaware that an `pandas.index` was supposed to be a valid input as well, I assumed this was only for our primary data containers (which had the bug that it did not support `pandas.DataFrame`).

Should be an easy fix, and I'll write a proper docstring. Hope everyone is watching this for how to write proper docstrings...

**Comment by fkiraly** [comment]
fix in #1582
